rest/data/interface.go

```diff
@@ -54,348 +35,10 @@ type Connector interface {
-	SetBannerTheme(string, *user.DBUser) error
-	// SetAdminBanner sets set the service flags in the system-wide settings document
-	SetServiceFlags(evergreen.ServiceFlags, *user.DBUser) error
-	RestartFailedTasks(amboy.Queue, model.RestartOptions) (*restModel.RestartResponse, error)
-	//RestartFailedCommitQueueVersions takes in a time range
-	RestartFailedCommitQueueVersions(opts model.RestartOptions) (*restModel.RestartResponse, error)
-	RevertConfigTo(string, string) error
-	GetAdminEventLog(time.Time, int) ([]restModel.APIAdminEvent, error)
-	MapLDAPGroupToRole(string, string) error
-	UnmapLDAPGroupToRole(string) error
-
-	// FindRecentTasks finds tasks that have recently finished.
-	FindRecentTasks(int) ([]task.Task, *task.ResultCounts, error)
-	FindRecentTaskListDistro(int) (*restModel.APIRecentTaskStatsList, error)
-	FindRecentTaskListProject(int) (*restModel.APIRecentTaskStatsList, error)
-	FindRecentTaskListAgentVersion(int) (*restModel.APIRecentTaskStatsList, error)
-	// GetHostStatsByDistro returns host stats broken down by distro
-	GetHostStatsByDistro() ([]host.StatsByDistro, error)
-
-	AddPublicKey(*user.DBUser, string, string) error
-	DeletePublicKey(*user.DBUser, string) error
-	GetPublicKey(*user.DBUser, string) (string, error)
-	UpdateSettings(*user.DBUser, user.UserSettings) error
-	SubmitFeedback(restModel.APIFeedbackSubmission) error
-
-	AddPatchIntent(patch.Intent, amboy.Queue) error
-
-	SetHostStatus(*host.Host, string, string) error
-	SetHostExpirationTime(*host.Host, time.Time) error
-
-	// TerminateHost terminates the given host via the cloud provider's API
-	TerminateHost(context.Context, *host.Host, string) error
-
-	// DisableHost disables the host, notifies it's been disabled,
-	// and clears and resets its running task.
-	DisableHost(context.Context, evergreen.Environment, *host.Host, string) error
-
-	CheckHostSecret(string, *http.Request) (int, error)
-
-	// CreatePod creates a new pod and returns the result of creating the pod.
-	CreatePod(restModel.APICreatePod) (*restModel.APICreatePodResponse, error)
-	// FindPodByID finds a pod by the given ID.
-	FindPodByID(id string) (*restModel.APIPod, error)
-	// FindPodByExternalID finds a pod by the given external identifier.
-	FindPodByExternalID(id string) (*restModel.APIPod, error)
-	// UpdatePodStatus updates a pod's status by ID.
-	UpdatePodStatus(id string, current, updated restModel.APIPodStatus) error
-	// CheckPodSecret checks that the ID and secret match the server's
-	// stored credentials for the pod.
-	CheckPodSecret(id, secret string) error
-
-	// FindProjectAliases queries the database to find all aliases (including or excluding those specified).
-	// Includes repo aliases if repoId is provided.
-	FindProjectAliases(string, string, []restModel.APIProjectAlias) ([]restModel.APIProjectAlias, error)
-	// CopyProjectAliases copies aliases from the first project for the second project.
-	CopyProjectAliases(string, string) error
```

Comment:
I didn't review this closely but if these functions don't have comments, can we move their interface comment to be the function comment?